tenant-auth/src/main/java/ai/vespa/hosted/auth/EndpointAuthenticator.java

```diff
@@ -1,36 +1,39 @@
 package ai.vespa.hosted.auth;
 
-import ai.vespa.hosted.api.ControllerHttpClient;
-import com.yahoo.config.provision.ApplicationId;
+import com.yahoo.config.provision.SystemName;
 import com.yahoo.security.KeyUtils;
 import com.yahoo.security.SslContextBuilder;
 import com.yahoo.security.X509CertificateUtils;
 
 import javax.net.ssl.SSLContext;
 import java.io.IOException;
 import java.io.UncheckedIOException;
-import java.net.URI;
 import java.net.http.HttpRequest;
 import java.nio.file.Files;
 import java.nio.file.Path;
-import java.nio.file.Paths;
 import java.security.NoSuchAlgorithmException;
 import java.security.PrivateKey;
 import java.security.cert.X509Certificate;
 import java.time.Instant;
 import java.util.Optional;
 
+import static ai.vespa.hosted.api.Properties.getNonBlankProperty;
+
 /**
- * Authenticates {@link HttpRequest}s against a hosted Vespa application based on mutual TLS.
+ * Authenticates against the hosted Vespa API using private key signatures, and against Vespa applications using mutual TLS.
  *
  * @author jonmv
  */
-public class Authenticator {
+public class EndpointAuthenticator implements ai.vespa.hosted.api.EndpointAuthenticator {
+
+    /** Don't touch. */
```

Comment:
"don't touch" means "do not call" or do not modify? Why do you make a constructor we're not supposed to call?